git_machete/git_operations.py

```diff
@@ -376,3 +375,24 @@ def load_branches(cli_ctxt: CommandLineContext) -> None:
         GitContext.committer_unix_timestamp_by_revision_cached[b] = int(committer_unix_timestamp_and_time_zone.split(' ')[0])
         if fetch_counterpart_stripped in GitContext.remote_branches_cached:
             GitContext.counterparts_for_fetching_cached[b_stripped] = fetch_counterpart_stripped
+
+
+def log_shas(cli_ctxt: CommandLineContext, revision: str, max_count: Optional[int]) -> List[str]:
+    opts = ([f"--max-count={str(max_count)}"] if max_count else []) + ["--format=%H", f"refs/heads/{revision}"]
+    return utils.non_empty_lines(popen_git(cli_ctxt, "log", *opts))
+
+
+# Since getting the full history of a branch can be an expensive operation for large repositories (compared to all other underlying git operations),
+# there's a simple optimization in place: we first fetch only a couple of first commits in the history,
+# and only fetch the rest if none of them occurs on reflog of any other branch.
```

Comment:
I'd rephrase this particular line to sth like:
```suggestion
# and only fetch the rest if needed.
```
to avoid machete-specific mention about reflogs.

Reply:
https://virtuslab.github.io/tips/#git/git-reflog ➕ https://github.com/VirtusLab/git-machete/blob/master/docs/blog-1/blog-1.md#appendix-fork-point---not-so-easy-business for details (although I admit the latter is pretty technical)